File: led_panel.py
#!/usr/bin/env python

import serial

class led_panel:
  def __init__(self, device):
    self.ser = serial.Serial(device, 9600)
    #print self.ser

  def close(self):
    self.ser.close()

  def plot(self, x, y, color):
    address = (y * 32) + x
    self.ser.write(chr(address >> 8))
    self.ser.write(chr(address & 0xff))
    self.ser.write(chr(color))
    self.ser.read(1)

  def page_flip(self):
    self.ser.write(chr(0xff))
    self.ser.read(1)

  def clear_draw_buffer(self):
    self.ser.write(chr(0xfe))
    self.ser.read(1)

  def copy_display_buffer(self):
    self.ser.write(chr(0xfd))
    self.ser.read(1)

  def shift_left(self):
    self.ser.write(chr(0xfc))
    self.ser.read(1)

  def shift_right(self):
    self.ser.write(chr(0xfb))
    self.ser.read(1)

  def shift_up(self):
    self.ser.write(chr(0xfa))
    self.ser.read(1)

  def shift_down(self):
    self.ser.write(chr(0xf9))
    self.ser.read(1)

File: nodejs/led_panel.js

// LED Panel node.js module - Copyright 2014-2016 by Michael Kohn
// Email: mike@mikekohn.net
//   Web: http://www.mikekohn.net/
//
// Control an RGB 32x16 LED panel with a Tessel 2.

var tessel = require('tessel');
var buffer = new Buffer(1);
var point = new Buffer(3);

module.exports =
{

  init: function()
  {
    port = tessel.port.A;
    uart = new port.UART({ baudrate: 9600 });
  },

  close: function()
  {
  },

  plot: function(x, y, color)
  {
    address = (y * 32) + x;
    point[0] = address >> 8;
    point[1] = address && 0xff;
    point[2] = color;
    uart.write(point);
    uart.on('data', function (data) { });
  },

  page_flip: function()
  {
    buffer[0] = 0xff;
    uart.write(buffer);
    uart.on('data', function (data) { });
  },

  clear_draw_buffer: function()
  {
    buffer[0] = 0xfe;
    uart.write(buffer);
    uart.on('data', function (data) { });
  },

  copy_display_buffer: function()
  {
    buffer[0] = 0xfd;
    uart.write(buffer);
    uart.on('data', function (data) { });
  },

  shift_left: function()
  {
    buffer[0] = 0xfc;
    uart.write(buffer);
    uart.on('data', function (data) { });
  },

  shift_right: function()
  {
    buffer[0] = 0xfb;
    uart.write(buffer);
    uart.on('data', function (data) { });
  },

  shift_up: function()
  {
    buffer[0] = 0xfa;
    uart.write(buffer);
    uart.on('data', function (data) { });
  },

  shift_down: function()
  {
    buffer[0] = 0xf9;
    uart.write(buffer);
    uart.on('data', function (data) { });
  },
};

var port;
var uart;


